google/cloud/sql/connector/connector.py

```diff
@@ -25,41 +25,125 @@
 from threading import Thread
 from typing import Any, Dict, Optional
 
-# This thread is used to background processing
-_thread: Optional[Thread] = None
-_loop: Optional[asyncio.AbstractEventLoop] = None
-_keys: Optional[concurrent.futures.Future] = None
-
-_instances: Dict[str, InstanceConnectionManager] = {}
-
 logger = logging.getLogger(name=__name__)
 
 
-def _get_loop() -> asyncio.AbstractEventLoop:
-    global _loop
-    if _loop is None:
-        _loop = asyncio.new_event_loop()
-        _thread = Thread(target=_loop.run_forever, daemon=True)
-        _thread.start()
-    return _loop
+class Connector:
+    """A class to configure and create connections to Cloud SQL instances.
 
+    :type ip_types: IPTypes
+    :param ip_types
+        The IP type (public or private)  used to connect. IP types
+        can be either IPTypes.PUBLIC or IPTypes.PRIVATE.
 
-def _get_keys(loop: asyncio.AbstractEventLoop) -> concurrent.futures.Future:
-    global _keys
-    if _keys is None:
-        _keys = asyncio.run_coroutine_threadsafe(generate_keys(), loop)
-    return _keys
+    :type enable_iam_auth: bool
+    :param enable_iam_auth
+        Enables IAM based authentication (Postgres only).
 
+    :type timeout: int
+    :param timeout:
+        The time limit for a connection before raising a TimeoutError.
+
+    """
 
-def connect(
-    instance_connection_string: str,
-    driver: str,
-    ip_types: IPTypes = IPTypes.PUBLIC,
-    enable_iam_auth: bool = False,
-    **kwargs: Any
-) -> Any:
-    """Prepares and returns a database connection object and starts a
-    background thread to refresh the certificates and metadata.
+    def __init__(
+        self,
+        ip_types: IPTypes = IPTypes.PUBLIC,
+        enable_iam_auth: bool = False,
+        timeout: int = 30,
+    ) -> None:
+        # This thread is used for background processing
+        self._thread: Optional[Thread] = None
+        self._loop: Optional[asyncio.AbstractEventLoop] = None
+        self._keys: Optional[concurrent.futures.Future] = None
```

Comment:
Setting these to None was necessary before because they were global state. I think we can initialize by default since they are inside the object now.